### ISSUE TITLE
[Proposal] Add a callback once the initialization is finished

### ISSUE THREAD

**Issue by LuisMiguelFilipe** [wobsoriano/pinia-shared-state#5]
Hello,

Would it be possible to add a callback for when the initialization is finished.

In my case, I would like to perform some action depending on the `state` of the store.

Does it make sense?

I can try to make a PR, if you agree with proposal 

Thanks

**Comment by wobsoriano** [comment]
Sure feel free to PR

**Comment by wobsoriano** [comment]
Closing as it is out of scope for now.

**Comment by BenceSzalai** [comment]
**pubkey/broadcast-channel** offers LeaderElection, which can be used to determine if there are no other (responsive) tabs (within a somewhat deterministic timeframe).

Outside the store test if the tab is Leader using LeaderElection as described [here](https://github.com/pubkey/broadcast-channel#using-the-leaderelection).
  - If the tab is leader there will be no information coming from other tabs. Continue business as usual: you are initialised (or will be as soon as you call `useXYStore()`).
  - If the tab is not leader, then add watches for your state variables. When they change you will know information has been received from other tabs.

Expect the other tab to be closed or crash at any time, so if the watcher does not fire within an expected period of time, test if you are Leader again and proceed accordingly. Continue this recheck until either the store variables are updated or you became Leader.

Since `useXYStore()` is sync as well as the initialisation of Pinia Plugins if you put the above code between the first `useXYStore()` call and any actual store usage, noone else can make changes to the state, so you can be sure that any state change is an indication of being synced from other tabs (unless you use other plugins that modify the state during store initialisation).

Indeed this could be built into the plugin. Then the watching of state vars could be replaced by the more direct event of receiving a message, so that would be more roboust.
